Implement getResourceCondition function

jan-di opened the pull request:
### Description of your changes

Adds new function to make it easier to access the conditions of observed resources. With that function, you can check a status in a one line if condition, making the code much more visible.

Previously:
```golang
{{ $networkStatus := "Unknown" }}
{{ if .observed.resources }}
  {{- with (index .observed.resources "network").resource.status.conditions }}
    {{ range . }}
      {{ if eq .type "Ready" }}
        {{ $networkStatus = .status }}
      {{ end }}
    {{ end }}
  {{ end }}
{{ end }}
```

Now:
```golang
{{ $networkStatus := (getObservedResourceCondition . "network" "Ready").Status }}
```

Fixes #19 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit tests for my change.(Examples for crossplane beta render)

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

jan-di commented:
@ezgidemirel @phisco Hi, it seems i'm not able to request  a review using GitHub (missing permissions?) so please consider this PR as ready to be reviewed :-)

phisco commented:
Thanks @jan-di! FYI: we usually already consider non-draft PRs as "ready for review" 😉

phisco commented on README.md:
```diff
@@ -100,11 +100,12 @@ about `crossplane beta render`.
+| [`randomChoice`](example/functions/randomChoice)                                 | Randomly selects one of a given strings                      |
+| [`toYaml`](example/functions/toYaml)                                             | Marshals any object into a YAML string                       |
+| [`fromYaml`](example/functions/fromYaml)                                         | Unmarshals a YAML string into an object                      |
+| [`getObservedResourceCondition`](example/functions/getObservedResourceCondition) | Helper function to retreive conditions of observed resources |
```
```suggestion
| [`getObservedResourceCondition`](example/functions/getObservedResourceCondition) | Helper function to retrieve conditions of observed resources |
```

phisco commented on example/functions/getObservedResourceCondition/README.md:
````diff
@@ -0,0 +1,39 @@
+
+```golang
+// Print whole condition
+{{ getObservedResourceCondition . "project" "Ready" | toYaml }}
````
just to avoid having to deserialize the whole request, would it make sense to have it as a function to lookup a specific condition from a resource:
```
{{ getResourceCondition .observed.resources["project"] "Ready" | toYaml }}
```
or maybe even from a list of conditions:
```
{{ getCondition .observed.resources["project"].status.conditions "Ready" | toYaml }}
```

sure this would mean having to have an if to check the resource is there before calling it, but on the other hand it would be pretty straight forward and also support getting conditions in the composite resource out of the box.

jan-di replied:
Yeah, I understand where you coming from. I thought of this function primarily as a syntactic sugar to cleanup the go templates but I think it makes sense to just give a resource as parameter, and give the responsibility to the user to ensure that input is actually a resource and not null/whatever.

jan-di commented:
@phisco  Updated the function:

- Now it takes a resource as input, and therefore is compatible with composed and the composite resource
- Switched order of parameters, to support pipelining the function, as it feels more natural
- Added a few more usage examples in the example files

phisco commented on function_maps.go:
```diff
@@ -30,6 +36,39 @@ var funcMaps = []template.FuncMap{
 		},
+
+		"getResourceCondition": func(ct string, res map[string]any) (*xpv1.Condition, error) {
+			resource, ok := res["resource"]
```
nit: this will definitely work, but could be worth using the same approach we already use in a few other places: https://github.com/crossplane/crossplane/blob/ac99fe06e5e69d9824ee6dd0056417098bffb1ea/cmd/crank/beta/trace/internal/resource/resource.go#L37

jan-di replied:
great, didn't know that this function exists :) Refactored it using fieldpath.Pave

phisco reviewed (approved):
Thanks @jan-di!
